### DISCUSSION TITLE
What is the empty Data entry on unhandled exceptions?

### DISCUSSION THREAD

**Discussion by ThomasArdal** [dotnet/wpf#8444] [Q&A] [answered]
I'm implementing the `UnhandledException` event in WPF running on both .NET 6 and .NET 8:

```csharp
AppDomain.CurrentDomain.UnhandledException += (sender, args) =>
    Log(args.ExceptionObject as Exception);
```

When throwing a new exception somewhere in my code, I get the `UnhandledException` event triggered as expected. But exceptions are decorated with an empty entry in the `Data` dictionary where the key is of type `System.Object`:

![image](https://github.com/dotnet/wpf/assets/563206/9df09b3c-e050-41f8-b795-250bce25b296)

Do anyone know why this happens? Is it something that can be disabled or is this intended to include some additional information or something?

**Comment by miloush** [comment, marked as answer]
https://github.com/dotnet/wpf/blob/ed058c1ab3f5594110731354794c5dfa0debdbd4/src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Threading/Dispatcher.cs#L2755-L2767

**Reply by ThomasArdal** [reply]
Thanks. So, it's actually some internal bookkeeping to see if an exception has already been handled? And it is (mis)using the Data dictionary for that?

**Reply by miloush** [reply]
Correct. I wouldn't necessarily call finding something you didn't put there yourself a misuse (it is writeable by any catch/finally on the stack and anybody else having a reference too), but I understand the mild annoyance.

**Reply by ThomasArdal** [reply]
It's not a problem that items are added to the dictionary. As you write, only an annoyance. Data items are indeed additional information about the exception. And this is somewhat an additional piece of information. It's just hard to know what information {object} is :smile:

I personally prefer keeping items in Data as additional information about an exception that is interesting for anyone who gets hold of a reference to the exception. But that's probably a matter of taste. I can see how this can be used for bookkeeping too, I just think that I would have named it differently. But I'm happy knowing what this is now.

I'll remove it on my side. Thank you for the quick feedback.
